packages/datetime/test/timePickerTests.tsx

```diff
@@ -209,6 +209,214 @@ describe("<TimePicker>", () => {
         assertTimeIs(timePicker.state.value, 0, 0, 0, 0);
     });
 
+    describe("Time range - minTime and maxTime props", () => {
+        it("if defaultValue is smaller minTime, minTime becomes initial time", () => {
+            renderTimePicker({
+                defaultValue: new Date(1995, 6, 30, 12, 30),
+                minTime: new Date(1995, 6, 30, 15, 30),
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+
+            assert.strictEqual(hourInput.value, "15");
+            assert.strictEqual(timePicker.state.value.getHours(), 15);
+        });
+
+        it("if defaultValue is is greater than maxTime, minTime becomes initial time", () => {
```

Comment:
this is not what i'd expect. i'd expect to clamp to nearest bound, so default greater than max should result in max time as initial time.